BlockEntity/BECharcoalPit.cs

```diff
@@ -255,35 +252,70 @@ BlockPos FindHoleInPit()
 
                 foreach (BlockFacing facing in BlockFacing.ALLFACES)
                 {
-                    BlockPos npos = bpos.AddCopy(facing);
+                    facing.IterateThruFacingOffsets(npos);  // This must be the first command in the loop, to ensure all facings will be properly looped through regardless of any 'continue;' statements
                     IWorldChunk chunk = Api.World.BlockAccessor.GetChunkAtBlockPos(npos);
                     if (chunk == null) return null;
 
                     Block nBlock = chunk.GetLocalBlockAtBlockPos(Api.World, npos);
 
-                    bool solid = nBlock.SideSolid[facing.Opposite.Index] || (nBlock is BlockMicroBlock && (chunk.GetLocalBlockEntityAtBlockPos(npos) as BlockEntityMicroBlock).sideAlmostSolid[facing.Opposite.Index]);
+                    bool solid = nBlock.GetLiquidBarrierHeightOnSide(facing.Opposite, npos) == 1 || nBlock.GetLiquidBarrierHeightOnSide(facing, bpos) == 1;

```

Comment:
This allows certain things like Iron Hatch Doors, Iron Trapdoors, and Beehive Kiln Doors which don’t have SideSolid set because they shouldn’t allow for things like torches to be placed on them to still count as a solid wall for the sake of a charcoal pit.  Not Iron Doors, though, since they are hard coded to have a window in the top half that liquid can flow through.